app/src/main/java/tech/relaycorp/gateway/domain/publicsync/CollectParcelsFromGateway.kt

```diff
@@ -37,17 +41,31 @@ class CollectParcelsFromGateway
             poWebClient.use {
                 poWebClient
                     .collectParcels(arrayOf(signer), streamingMode)
+                    .retry(Long.MAX_VALUE) { e ->
+                        if (keepAlive && e is ServerException) {
```

Comment:
I think we shouldn't retry if the failure is due to a protocol violation (`ServerBindingException`): Retrying in a few seconds won't change anything. For that reason, I'd suggest to retry if the exception is a `ServerConnectionException`.

Now, this raises the question of how to handle protocol violations. In an ideal world, I think we should notify the user about it. However, since we'll be operating the initial server, I think that's very unlikely so we should just log that event.